plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/ui/GcpLocalRunTab.java

```diff
@@ -403,7 +406,7 @@ Path getServiceAccountKeyPath() {
     String projectId = projectSelector.getSelectProjectId();
     String filename = "app-engine-default-service-account-key-" //$NON-NLS-1$
         + projectId + ".json"; //$NON-NLS-1$
-    return Paths.get(System.getProperty("user.home")).resolve(filename); //$NON-NLS-1$
+    return Paths.get(Platform.getConfigurationLocation().getURL().getPath()).resolve(filename);
```

Comment:
The convention seems to be to create a directory to hold files, usually reverse-DNS-based or bundle symbolic name. Since we could generate non-AppEngine keys, perhaps just `com.google.cloud.tools.eclipse`?

Reply:
`Platform.getConfigurationLocation()` already gives `.metadata/.plugins/org.eclipse.pde.core/pde-junit/`.

Reply:
Oh, wait. That was in a testing.

Reply:
I think it'll still be under `org.eclipse.pde.core`.

Reply:
It will be the same as the `osgi.configuration.area` property.  In my IDE instance, for example (_Help > About > Configuration_), it's `file:/usr/local/installs/e4-dev/Eclipse.app/Contents/Eclipse/configuration/`.

Reply:
And for completeness, the contents:
```
$ ls -1 /usr/local/installs/e4-dev/Eclipse.app/Contents/Eclipse/configuration/
config.ini
org.eclipse.core.runtime/
org.eclipse.e4.ui.css.swt.theme/
org.eclipse.equinox.app/
org.eclipse.equinox.simpleconfigurator/
org.eclipse.equinox.source/
org.eclipse.help.base/
org.eclipse.oomph.setup/
org.eclipse.osgi/
org.eclipse.ui.intro.universal/
org.eclipse.update/
org.eclipse.wst.server.discovery/
```

Reply:
It *might* be worth putting project keys in project-specific directories, rather than encoding the project ID in the file name.  We may create other project-specific files and this would be a place to put them?

Reply:
Updated to create a key under `com.google.cloud.tools.eclipse`.

I'm sticking to putting a project ID in a filename rather than creating a directory. I think it's possible directories are YAGNI.